### ISSUE TITLE
tags._linux_platforms breaks when _PYTHON_HOST_PLATFORM is set

### ISSUE THREAD

**Issue by zware** [pypa/packaging#259]
When using a dirty hack to build custom platform-specific wheels for Alpine (setting `_PYTHON_HOST_PLATFORM` to "alpine", which `distutils.util.get_platform()` returns untouched), `packaging.tags._linux_platforms()` raises a `ValueError: not enough values to unpack (expected 2, got 1)` on `_, arch = linux.split("_", 1)`.  The simplest solution would be to use `*_, arch = ...` instead, but that could cause unexpected behavior if `distutils.util.get_platform()` returns "x86_64"; I'm not sure if that actually matters, though :)

We're fixing this on our side by just using "linux_alpine" for our tag instead of just "alpine", but this might want a fix here anyway since it's very dependent on the exact format of the response from `get_platform()`.

**Comment by brettcannon** [comment]
It's true it's dependent on the format, but the fact you're using "a dirty hack" doesn't make that a bad thing. 😉 

For reference, the line in question is:

https://github.com/pypa/packaging/blob/d49fdc500a7c057b71851847ff8d7cc92865bcf2/packaging/tags.py#L642

Honestly I wish there was a better way to get the architecture that was more reliable. Any ideas, @zware ?

**Comment by zware** [comment]
Right; we were not surprised to be broken for relying on this, but were disappointed that we had it merged for about a day before `pip` updated and broke it :).  We have the fix on our side in place already, so if you decide that this is fine as is here, we'll be OK as long as you continue to use `distutils.util.get_platform()` ;)

Unfortunately, I don't really have a better suggestion for getting the architecture, especially considering cross-compilation (which is the real purpose of `_PYTHON_HOST_PLATFORM`, as far as it goes).  I don't know if `packaging` cares about that case, though, and in fact know next to nothing at all about the `packaging` package :)

**Comment by brettcannon** [comment]
It's more about whether there is a more reliable way in the stdlib that I'm not thinking off to get the appropriate detail that we are grabbing via packaging.tags.
